workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/QueueListener.java

```diff
@@ -124,6 +155,137 @@ public void listener(
 		}
 	}
 
+	private void _assignUser(
+			String accountName, String contactEmailAddress,
+			String contactRoleName, String externalReferenceCode)
+		throws Exception {
+
+		if (contactRoleName.equals("Partner Manager")) {
+			String pmRoleName = "Partner Manager (PM)";
+
+			_assignUserToAccount(
+				accountName, contactEmailAddress, externalReferenceCode);
+			_assignUserToAccountRole(
+				externalReferenceCode, contactEmailAddress,
+				"[Account] " + pmRoleName);
+			_assignUserToRegularRole(contactEmailAddress, pmRoleName);
+		}
+
+		if (contactRoleName.equals("Partner Member")) {
+			String pmuRoleName = "Partner Marketing User (PMU)";
+			String psuRoleName = "Partner Sales User (PSU)";
+
+			_assignUserToAccount(
+				accountName, contactEmailAddress, externalReferenceCode);
+			_assignUserToAccountRole(
+				externalReferenceCode, contactEmailAddress,
+				"[Account] " + pmuRoleName);
+			_assignUserToAccountRole(
+				externalReferenceCode, contactEmailAddress,
+				"[Account] " + psuRoleName);
+			_assignUserToRegularRole(contactEmailAddress, pmuRoleName);
+			_assignUserToRegularRole(contactEmailAddress, psuRoleName);
+		}
+	}
+
+	private void _assignUserToAccount(
+		String accountName, String contactEmailAddress,
+		String externalReferenceCode) {
+
+		if (_log.isInfoEnabled()) {
+			_log.info(
+				StringBundler.concat(
+					"Assigning user ", contactEmailAddress, " to account ",
+					externalReferenceCode, " with name ", accountName));
+		}
+
+		_post(
+			"",
+			StringBundler.concat(
+				"/o/headless-admin-user/v1.0/accounts",
+				"/by-external-reference-code/", externalReferenceCode,
+				"/user-accounts/by-email-address/", contactEmailAddress));
+	}
+
+	private void _assignUserToAccountRole(
+		String externalReferenceCode, String contactEmailAddress,
+		String accountRoleName) {
+
+		long accountRolesId = _getAccountRolesId(
+			externalReferenceCode, accountRoleName);
+
+		if (accountRolesId <= 0) {
+			return;
+		}
+
+		if (_log.isInfoEnabled()) {
+			_log.info(
+				StringBundler.concat(
+					"Assigning user ", contactEmailAddress, " to account role ",
+					accountRoleName));
+		}
+
+		_post(
+			"",
+			StringBundler.concat(
+				"/o/headless-admin-user/v1.0/accounts",
+				"/by-external-reference-code/", externalReferenceCode,
+				"/account-roles/", accountRolesId,
+				"/user-accounts/by-email-address/", contactEmailAddress));
+	}
+
+	private void _assignUserToRegularRole(
+		String contactEmailAddress, String regularRoleName) {
```

Comment:
this can just be called "name" since there aren't any other "names" and the method name provides context about it